xbmc/cores/AudioEngine/Sinks/AESinkAUDIOTRACK.cpp

```diff
@@ -594,45 +590,25 @@ void CAESinkAUDIOTRACK::GetDelay(AEDelayStatus& status)
   // and add head_pos which wrapped around, e.g. 0x0001 0000 0000 -> 0x0001 0000 0004
   m_headPos = (m_headPos & UINT64_UPPER_BYTES) | (uint64_t)head_pos;
 
-  // head_pos does not necessarily start at the beginning
-  if (m_offset == -1 && m_at_jni->getPlayState() == CJNIAudioTrack::PLAYSTATE_PLAYING)
-  {
-    m_offset = m_headPos;
-  }
-
-  if (m_offset != -1 && (uint64_t) m_offset > m_headPos)
-  {
-    CLog::Log(LOGDEBUG, "You did it wrong man - fully wrong! offset %lld head pos %llu", m_offset, m_headPos);
-    m_offset = 0;
-  }
-
-  // we might not yet be running here, but we need m_offset to track first PT fillup
-  uint64_t normHead_pos = m_headPos;
-  if (m_offset > 0)
-    m_headPos -= m_offset;
-
-  if (m_passthrough && !m_info.m_wantsIECPassthrough)
-  {
-    if (m_extTimer.MillisLeft() > 0)
-    {
-      const double d = GetMovingAverageDelay(GetCacheTotal());
-      status.SetDelay(d);
-      return;
-    }
-  }
-
-  double gone = (double) normHead_pos / (double) m_sink_sampleRate;
+  double gone = (double)m_headPos / (double)m_sink_sampleRate;
```

Comment:
Can you replace the C style casts with C++ style casts as you already touched the line?

Reply:
Yes - will do. Thanks.